core/src/UserSettingsDialog.html

```diff
@@ -0,0 +1,294 @@
+    let getTranslation = getContext('getTranslation');
+    let dialogHeader = TOP_NAV_DEFAULTS.userSettingsDialog.dialogHeader;
+    let saveBtn = TOP_NAV_DEFAULTS.userSettingsDialog.saveBtn;
+    let dismissBtn = TOP_NAV_DEFAULTS.userSettingsDialog.dismissBtn;
+
+    onMount(() => {
+        const usLocalization = LuigiConfig.getConfigValue('settings.userSettings.userSettingsDialog');
+        if (usLocalization) {
+            dialogHeader = usLocalization.dialogHeader ? usLocalization.dialogHeader : dialogHeader;
+            saveBtn = usLocalization.saveBtn ? usLocalization.saveBtn : saveBtn;
+            dismissBtn = usLocalization.dismissBtn ? usLocalization.dismissBtn : dismissBtn;
+        }
+        let key;
+        for (let k in userSettingsData[0]) {
+            key = k;
+            break;
+        }
+        LuigiConfig.readUserSettings().then(userSettingsFromStorage => {
+            previousUserSettings = JSON.parse(JSON.stringify(userSettingsFromStorage));
+            if (userSettingsFromStorage === null) {
+                storedUserSettings;
+            } else {
+                storedUserSettings = userSettingsFromStorage;
+            }
+            openEditor([key, userSettingsData[0][key]]);
+        });
+    });
+
+    function openEditor(category, event) {
+        if (event) {
+            document.querySelectorAll('.lui-us-list .fd-nested-list__link').forEach(elem => {
+                elem.classList.remove('is-selected');
+            });
+            const link = event.target.closest('.fd-nested-list__link');
+            link.classList.add("is-selected");
+            if (window.innerWidth < CSS_BREAKPOINTS.desktopMinWidth) {
+                closeNavOnCategoryClickMobile();
+            }
+        } else {
+            document.querySelectorAll('.lui-us-list .fd-nested-list__link')[0].classList.add("is-selected");
+        }
+        userSettingsSchema = category;
+        displayEditor = true;
+    }
+
+    const updateSettingsObject = (event) => {
+        storedUserSettings = event.detail.userSettings;
+    }
+
+    function storeUserSettings() {
+        LuigiConfig.storeUserSettings(storedUserSettings, previousUserSettings).then(() => {
+            dispatch('close');
+        });
+    }
+
+    function closeNavOnCategoryClickMobile() {
+        toggleNavMobile();
+    }
+
+    function toggleNavMobile() {
+        document.querySelector('.lui-usersettings-dialog').classList.toggle('usersettings-leftNavVisible');
+    }
+    function hasOpenUIicon(node) {
+        return NavigationHelpers.isOpenUIiconName(node.icon);
+    }
+
+    export function handleKeydown(event) {
+        if (event.keyCode === KEYCODE_ESC) {
+            dispatch('close');
+        }
+    }
+</script>
+
+<style>
+    .dialog-size {
+        width: 60vw;
+        height: 60vh;
+    }
+
+    .user-settings__body {
+        position: relative;
+        padding-top: 0;
+        padding-bottom: 0;
+    }
+
+    .user-settings__leftnav {
```

Comment:
Would that be easier to debug the HTML if we rename the classes by using prefix "luigi"?e.g. .lui-usersettings__body, .lui-usersettings__left-nav